scripts/main/albums.js

```diff
@@ -154,78 +154,74 @@ albums.isShared = function (albumID) {
 	return found;
 };
 
+/**
+ * @param {?string} albumID
+ * @returns {(null|Album|TagAlbum|SmartAlbum)}
+ */
 albums.getByID = function (albumID) {
-	// Function returns the JSON of an album
+	if (albumID == null) return null;
+	if (!albums.json) return null;
+	if (!albums.json.albums) return null;
 
-	if (albumID == null) return undefined;
-	if (!albums.json) return undefined;
-	if (!albums.json.albums) return undefined;
-
-	let json = undefined;
-
-	let func = function () {
-		if (this.id === albumID) {
-			json = this;
-			return false; // stop the loop
-		}
-		if (this.albums) {
-			$.each(this.albums, func);
-		}
-	};
+	if (albums.json.smart_albums.hasOwnProperty(albumID)) {
+		return albums.json.smart_albums[albumID];
+	}
 
-	$.each(albums.json.albums, func);
+	let result = albums.json.tag_albums.find((tagAlbum) => tagAlbum.id === albumID);
+	if (result) {
+		return result;
+	}
 
-	if (json === undefined && albums.json.shared_albums !== null) $.each(albums.json.shared_albums, func);
+	result = albums.json.albums.find((album) => album.id === albumID);
+	if (result) {
+		return result;
+	}
 
-	if (json === undefined && albums.json.smart_albums !== null) $.each(albums.json.smart_albums, func);
+	result = albums.json.shared_albums.find((album) => album.id === albumID);
+	if (result) {
+		return result;
+	}
 
-	return json;
+	return null;
 };
 
+/**
+ * Deletes an album by ID.
+ *
+ * This function is only ever invoked for top-level albums, so it doesn't
+ * need to walk the albums tree.
+ *
+ * TODO: Check above statement.
+ *
+ * The method is called by {@link album.delete} after the album has
+ * successfully been deleted at the server-side.
+ * How does this match the statement above?
```

Comment:
I fixed the JSdoc comment of the method in: https://github.com/LycheeOrg/Lychee-front/pull/282/commits/5f3bc868d4c6d3cba527f0b80b1b612be1567356

> Some of the other functions here could probably be simplified for this reason -- they don't need to be recursive anymore.

Do you have any particular methods in mind? If yes, I happy to hear about it and see what we can do to simplify them. If not, I would resolve this conversation.

Reply:
Let me see... `albums.getByID` used to be recursive but it no longer is on this branch, so that's taken care of... `albums.isShared` is still recursive though it no longer needs to be.